definitions/types/azure-subnet-id.json

```diff
@@ -0,0 +1,13 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "type": "string",
+  "title": "Azure Subnet Resource ID",
+  "description": "Azure Subnet Resource ID",
+  "examples": [
+    "/subscriptions/12345678-1234-1234-abcd-1234567890ab/resourceGroups/resource-group-name/providers/Microsoft.Network/virtualNetworks/network-name/subnets/subnet-name"
+  ],
+  "pattern": "^\/subscriptions\/[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}\/resourceGroups\/[a-zA-Z0-9-_.()]{0,89}[a-zA-Z0-9-_()]\/providers\/Microsoft\\.[a-zA-Z]+\/[a-zA-Z]+\/[0-9A-Za-z-]+\/subnets\/[a-zA-Z0-9][a-zA-Z0-9_.-]{0,78}[a-zA-Z0-9_]{0,1}$",
```

Comment:
It always blows my mind that `my...9()` is a valid `resourceGroup` ID.

For the very last block `[a-zA-Z0-9_]{0,1}` do we want that to be `[a-zA-Z0-9_]+` Usually when I see things at the end like that it means "it has to end with this" and `{0,1}` means it _can_. It looks like the subnet can't end with a period yeah? If so, then I think we want the `+` there.

The rest looks pretty good to me!

Reply:
So my logic with `[a-zA-Z0-9][a-zA-Z0-9_.-]{0,78}[a-zA-Z0-9_]{0,1}` is:

- Subnet names can start with letters or numbers
- Subnet names can end with a number, letter, or underscore
- Subnet names can contain numbers, letters, underscores, periods, and hyphens
- Subnet names can be between 1-80 characters in length

So, since subnet names can't end with a period or hyphen, that's why I set it up the way I did. Do you have a better regex to fit those params?

Reply:
Ahh I missed that first block as well. I think this is what you're looking for

[a-zA-Z0-9]+[a-zA-Z0-9_.-]{0,78}[a-zA-Z0-9_]+